chart/tests/test_flower.py

```diff
@@ -358,18 +358,18 @@ def test_defaults(self):
         "ports, expected_ports",
         [
             (
-                [{"name": "{{ .Release.Name }}", "protocol": "UDP", "port": "{{ .Values.ports.flowerUI }}"}],
-                [{"name": "RELEASE-NAME", "protocol": "UDP", "port": 5555}],
+                [{"protocol": "UDP", "port": "{{ .Values.ports.flowerUI }}"}],
+                [{"protocol": "UDP", "port": 5555}],
             ),
-            ([{"name": "only_sidecar", "port": "sidecar"}], [{"name": "only_sidecar", "port": "sidecar"}]),
+            ([{"port": "sidecar"}], [{"port": "sidecar"}]),
             (
                 [
-                    {"name": "flower-ui", "port": "{{ .Values.ports.flowerUI }}"},
-                    {"name": "sidecar", "port": 80},
+                    {"port": "{{ .Values.ports.flowerUI }}"},
+                    {"port": 80},
```

Comment:
```suggestion
                   {"name": "flower-ui", "port": "{{ .Values.ports.flowerUI }}"},
                   {"name": "sidecar-{{ .Release.Name }}", "port": 80},
```

`name` is required if there is more than 1 ServicePort (sorry, deepest I can link you): https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/#ServiceSpec

Also add back a test that `name` is templated.

Reply:
This is a test for NetworkPolicy, not Service. In NetworkPolicy, we use [https://github.com/yannh/kubernetes-json-schema/blob/master/v1.21.0/networkpolicyport-networking-v1.json](NetworkPolicyPort), not [ServicePort](https://github.com/yannh/kubernetes-json-schema/blob/master/v1.21.0/serviceport-v1.json), so `name` field is illegal here.

Reply:
Ah, whoops!